packages/web/src/components/buy-audio-modal/components/StripeBuyAudioButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from 'react'
+
+import {
+  OnRampProvider,
+  buyAudioSelectors,
+  accountSelectors,
+  buyAudioActions
+} from '@audius/common'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { useModalState } from 'common/hooks/useModalState'
+import { OnRampButton } from 'components/on-ramp-button'
+import { createStripeSession } from 'services/audius-backend/BuyAudio'
+
+const { getAccountUser } = accountSelectors
+const { getAudioPurchaseInfo } = buyAudioSelectors
+const { calculateAudioPurchaseInfo } = buyAudioActions
+
+// TODO: Replace this with Stripe npm package when available
+// @ts-ignore
+const StripeOnRamp = window.StripeOnramp
+
+export const StripeBuyAudioButton = () => {
+  const dispatch = useDispatch()
+  const [, setIsStripeModalVisible] = useModalState('StripeOnRamp')
+  const user = useSelector(getAccountUser)
+
+  const purchaseInfo = useSelector(getAudioPurchaseInfo)
+  const amount =
+    purchaseInfo?.isError === false
+      ? purchaseInfo.estimatedSOL.uiAmountString
```

Comment:
nit:
is it possible that the purchaseInfo amount does not depend on the isError state in the components. there is also a similar logic in the CoinbaseBuyAudioButton. was wondering if we could give the store amount the right state when isError is not false.

Reply:
Yeah this is for typescript to infer types. I'm using `isError` as a discriminator between the "error" type and the successful purchase info. Do you think it makes more sense to use empty objects instead?